tools/topology/topology1/sof/pipe-volume-demux-playback.m4

```diff
@@ -69,10 +69,10 @@ W_MUXDEMUX(0, 1, PIPELINE_FORMAT, 2, 2, SCHEDULE_CORE,
 	LIST(`         ', concat(`DEMUX', PIPELINE_ID)))
 
 # Low Latency Buffers
-W_BUFFER(0, COMP_BUFFER_SIZE(2,
+W_BUFFER(0, COMP_BUFFER_SIZE(DAI_PERIODS,
```

Comment:
I think the topology is good but the volume control is wrong here.
```c
	if (sinkb->stream.size < dev->ipc_config.periods_sink * sink_period_bytes) {
		comp_err(dev, "volume_prepare(): sink buffer size %d is insufficient < %d * %d",
			 sinkb->stream.size, dev->ipc_config.periods_sink, sink_period_bytes);
		ret = -ENOMEM;
		goto err;
	}
```
volume should only check if the sink buffer can hold 1 period of data.
@singalsu can you check volume and other components for this check. Thanks.

Reply:
Ack and thanks @lgirdwood @singalsu for review. This issue was reported as P1 for ADL with sof-tgl-max98357a-rt5682 tplg, will close this PR once a formal solution is provided.

Reply:
@lgirdwood Thanks, I'll start to work with fix for all components with same issue.

Reply:
Almost done... I think I'll leave SRC and ASRC for later. They do not in all cases survive with one period sink.

Reply:
@yongzhi1 Could you please try if PR #4659 works the issue with alsa_conformance_test.